firedrake/checkpointing.py

```diff
@@ -779,7 +814,7 @@ def _save_function_space_topology(self, tV):
                     topology_dm.setName(base_tmesh_name)
 
     @PETSc.Log.EventDecorator("SaveFunction")
-    def save_function(self, f, idx=None, name=None):
+    def save_function(self, f, idx=None, name=None, timetag=None):
```

Comment:
How is `timetag` different to `idx`? `idx` is described in the docstring as an "optionally timestepping index" which seems quite similar.
Also please add this new argument to the docstring.

Reply:
`timetag` is for the actual time (`float`), I think. I think `timestamp` (or simply `t`) would be less confusing.

Reply:
@ksagiyam is correct. `t` might be less confusing.

Reply:
I have now changed `timetag` to `t` to be less confusing. Following @ksagiyam's suggestion I have now added `timestep` too.